src/main/java/org/amahi/anywhere/adapter/NavigationDrawerAdapter.java

```diff
@@ -40,14 +57,44 @@ public NavigationDrawerViewHolder onCreateViewHolder(ViewGroup parent, int viewT
         return new NavigationDrawerViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.view_navigation_item, parent, false));
     }
 
+    private Drawable getServerBubble(ArrayList<String> serverList, int position) {
+        //get first letter of each String item
+        String firstLetter = String.valueOf(serverList.get(position).charAt(0));
+
+        ColorGenerator generator = ColorGenerator.MATERIAL; // or use DEFAULT
+        // generate random color
+        int color = generator.getColor(serverList.get(position));
+        //int color = generator.getRandomColor();
+
+        TextDrawable drawable = TextDrawable.builder()
+            .buildRound(firstLetter, color); // radius in px
+
+        return drawable;
+    }
+
     @Override
     public void onBindViewHolder(NavigationDrawerViewHolder holder, int position) {
+
+        if (mServerName != null) {
+            holder.server_bubble.setImageDrawable(getServerBubble(mServerName, position));
+            holder.titleShare.setText(mServerName.get(position));
+            return;
+        }
+
         holder.titleShare.setText(getNavigationName(mContext, position));
+        holder.server_bubble.setImageResource(R.drawable.ic_app_logo_shadowless);
+        holder.server_bubble.setScaleType(ImageView.ScaleType.CENTER_CROP);
+
+
     }
 
     @Override
     public int getItemCount() {
-        return navigationItems.size();
+        if (mNavigationItems != null) {
```

Comment:
This may be better.
`return mNavigationItems!=null?mNavigationItems.size():mServerName.size();`
Also is there are cases when we are getting null list?

Reply:
I understand your concern for the null list but here, we are changing the list items from the same area with either server Names or with selected server option (shares, apps).